client/webpack.config.js

```diff
@@ -113,6 +114,16 @@ function filterEntrypoints( entrypoints ) {
 	return allowed;
 }
 
+function addHotMiddlewareIfDev( entry ) {
+	if ( isDevelopment ) {
+		return Object.entries( entry ).reduce( ( acc, [ key, value ] ) => {
+			acc[ key ] = [ 'webpack-hot-middleware/client', ...value ];
+			return acc;
+		}, {} );
+	}
+	return entry;
```

Comment:
Nit, but I generally recommend the other way around:

```suggestion
	if ( ! isDevelopment ) {
		return entry;
	}

	return Object.entries( entry ).reduce( ( acc, [ key, value ] ) => {
		acc[ key ] = [ 'webpack-hot-middleware/client', ...value ];
		return acc;
	}, {} );
```

That way the actual logic is one less tab nested, and that can arguably improve the readability for some folks.

Reply:
Fixed